pkg/virt-launcher/virtwrap/util/libvirt_helper.go

```diff
@@ -183,6 +183,33 @@ func StartLibvirt(stopChan chan struct{}) {
 	// container.
 
 	go func() {
+		// libvirtd needs access to /dev/null and /dev/urandom
+		// Ensure both are there and ready to be used, panic if they are still not after 1 second
+		devNodes := []string{"/dev/null", "/dev/urandom"}
+		for i := 1; i <= 10; i++ {
+			var err error
+			var fd *os.File
+			for _, node := range devNodes {
+				fd, err = os.Open(node)
+				if err == nil {
+					fd.Close()
+				} else {
+					break
+				}
+			}
+			if err == nil {
+				break
+			} else {
+				log.Log.Reason(err).Warningf("one or more dev nodes were not ready (attempt %d/10)", i)
+				if i < 10 {
+					time.Sleep(100 * time.Millisecond)
```

Comment:
I'm confused by this. Is 1 second really the best value here?

What populates these device nodes? I'd assume that if the container engine was responsible, this would be done before any processes started.

If there's really a race condition with the devices not being present at container start, then wouldn't 1 second be a bit aggressive? Think high load clusters.

Reply:
We're not sure what's causing this, so the timing is a bit tricky to figure out...
I'm open to any value here, I assume by "aggressive" you mean short, so maybe 200ms? Or more?

Reply:
I'm more focused on the 1 second overall timeout. doing a rapidfire test doesn't hurt here because it's local to the container.